scipy/linalg/tests/test_basic.py

```diff
@@ -495,35 +497,41 @@ class TestSolve:
     def setup_method(self):
         np.random.seed(1234)
 
-    def test_20Feb04_bug(self):
-        a = [[1, 1], [1.0, 0]]  # ok
-        x0 = solve(a, [1, 0j])
-        assert_array_almost_equal(dot(a, x0), [1, 0])
+    @array_api_compatible
+    def test_20Feb04_bug(self, xp):
+        a = xp.asarray([[1, 1], [1.0, 0]], dtype=xp.complex128)  # ok
+        x0 = solve(a, xp.asarray([1, 0j]))
+        xp_assert_close(a @ x0, xp.asarray([1, 0], dtype=xp.complex128))
```

Comment:
I don't think SciPy need to test for other inputs since the logic is elsewhere so this is potentially a break that we can't do anything about. Hence in my opinion it doesn't need all the xp_assert_closes or xp.asarrays